src/SqlAsyncCollector.cs

```diff
@@ -237,14 +239,32 @@ private async Task UpsertRowsAsync(IEnumerable<T> rows, SqlAttribute attribute,
         /// to upsert to and returns the extra property names in a List.
         /// </summary>
         /// <param name="columns"> The columns of the table to upsert to </param>
+        /// <param name="rowItem"> Sample row used to get the column names when item is a JObject </param>
         /// <returns>List of property names that don't exist in the table</returns>
-        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns)
+        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns, T rowItem)
         {
+            if (typeof(T) == typeof(JObject))
+            {
+                Dictionary<string, string> dictObj = (rowItem as JObject).ToObject<Dictionary<string, string>>();
+                return dictObj.Keys.Where(prop => !columns.ContainsKey(prop))
+                .Select(prop => prop);
+            }
             return typeof(T).GetProperties().ToList()
                 .Where(prop => !columns.ContainsKey(prop.Name))
                 .Select(prop => prop.Name);
         }
 
+        private static IEnumerable<string> GetColumnNamesFromPOCO(T row, bool bracketed = false)
```

Comment:
I still don't see bracketed ever being set to true. Can you point me to where that happens? If not then just remove this parameter.

Reply:
I removed the one place it was being used since it's not actually being referred anywhere -> https://github.com/Azure/azure-functions-sql-extension/blob/a61c8478bf6f256842c0b9792135a3ed5eeff717/src/SqlAsyncCollector.cs#L399

Reply:
I'm not really sure what you're saying. I'm pointing out that you have a parameter `bracketed` with a default of false, but never actually call this method in a way that sets it to true. So you can just remove the parameter since it's not doing anything.

(assuming of course this isn't a bug and you actually don't care about this being bracketed ever...)